lib/users.ts

```diff
@@ -140,3 +210,100 @@ export const getUsers = async (ability: UserAbility) => {
     throw e;
   }
 };
+
+/**
+ * Update and overwrite existing User active status
+ * @param userID id of the user to be updated
+ * @param active activate or deactivate user
+ * @returns User
+ */
+export const updateActiveStatus = async (ability: UserAbility, userID: string, active: boolean) => {
+  try {
+    const canManageUsers = ability?.can("update", "User") ?? false;
+    if (!canManageUsers) throw new AccessControlError("Access Denied");
+
+    checkPrivileges(ability, [{ action: "update", subject: "User" }]);
+    const user = await prisma.user.update({
+      where: {
+        id: userID,
+      },
+      data: {
+        active: active,
+      },
+      select: {
+        id: true,
+        active: true,
+        email: true,
+      },
+    });
+
+    if (!active && user.email) {
+      sendDeactivationEmail(user.email);
+    }
+
+    return user;
+  } catch (error) {
+    logMessage.error(error as Error);
```

Comment:
Need to add a logEvent for the AccessControlError and rethrow (error) to bubble up the scope.